### ISSUE TITLE
Bert similarity score high for non semantic similar sentences .

### ISSUE THREAD

**Issue by ajbot2019** [google-research/bert#904]
hidden_reps, cls_head = bert_model(token_ids, attention_mask = attn_maskT,token_type_ids = seg_idsT)

Is the output of the token embedding normalized in berth ? just like how its normalized in google universal sentence encoding where we use just np.inner to get the similarity between 2 vector  ?
The problem here is  while calculating  similarity between 2 sentences USC give accurate scores than berth for semantic similar and non similar sentences  .

This is how i got the sentence vector 

 qtokens=tokenizer.tokenize(ques)
  qtokens = ['[CLS]'] + qtokens + ['[SEP]']
  T=10
  qpadded_tokens=qtokens +['[PAD]' for _ in range(T-len(qtokens))]
  qattn_mask=[ 1 if qtoken != '[PAD]' else 0 for qtoken in qpadded_tokens  ]
  qseg_ids=[0 for _ in range(len(qpadded_tokens))]
  qtoken_ids =tokenizer.convert_tokens_to_ids(qpadded_tokens)
  qtoken_idsT = torch.tensor(qtoken_ids).unsqueeze(0)
  qattn_maskT = torch.tensor(qattn_mask).unsqueeze(0) 
  qseg_idsT   = torch.tensor(qseg_ids).unsqueeze(0)
  qhidden_reps, qcls_head = bert_model(qtoken_idsT,attention_mask=qattn_maskT,token_type_ids= qseg_idsT)
  sentence 1= torch.mean(qhidden_reps[0],1)

#calculating the cosine similarity using pytorch cos function :
cos = torch.nn.CosineSimilarity(dim=0)
cos(sentence 1,, sentence 2)

sentences 1 | sentence 2 |cosine score
what is your age?   How old are you?    tensor(0.9897, grad_fn=<DivBackward0>)
what is your age?    I am very young     tensor(0.9472, grad_fn=<DivBackward0>)
what is your age?    Today is monday     tensor(0.9396, grad_fn=<DivBackward0>)
what is your age?    what is your age?      tensor(1., grad_fn=<DivBackward0>)
 what is your age?       How are you?         tensor(0.9260, grad_fn=<DivBackward0>).

As you can see i used the same examples provided in google USC colab 
the similarity  for these are very high even though they are not similar is any way .USC gave better score than BERT for non semantic similar sentences.
what is your age?  and   Today is monday
what is your age?   and     How are you?
